### PR TITLE
Update Open API guidance

### PR DESCRIPTION
**Pull request by MassimoC**
Add more information as #51

### CODE REVIEW

File: maturity-level-one/README.md [hunk 1]
```diff
@@ -150,20 +150,28 @@ Every API should have documentation in the OpenAPI format. If you want to genera
...
-}
+        /// <summary>
+        /// Get the profiles of the players
+        /// </summary>
+        /// <param name="topPlayersOnly">Indicates whether to return the top players only</param>
+        /// <remarks>Operation description here</remarks>
+        /// <returns>Return a list of Players</returns>
+        [HttpGet(Name = "Players_GetPlayers")]
+        [SwaggerResponse((int)HttpStatusCode.OK, "List of players")]
+        [SwaggerResponse((int)HttpStatusCode.InternalServerError, "API is no
```
Comment (tomkerkhove) [inline maturity-level-one/README.md]: Attribute seems to be cut of

File: maturity-level-one/README.md [hunk 2]
```diff
@@ -150,20 +150,28 @@ Every API should have documentation in the OpenAPI format. If you want to genera
...
-[Route("health")]
-[SwaggerOperation("Health_Get")]
-[SwaggerResponse(HttpStatusCode.OK, "API is up & running")]
-[SwaggerResponse(HttpStatusCode.InternalServerError, "API is not available")]
-public IHttpActionResult Get()
-{
-    return Ok();
-}
+        /// <summary>
+        /// Get the profiles of the players
```
Comment (tomkerkhove) [inline maturity-level-one/README.md]: Pretty long Operation name, no?

```suggestion
        /// Get Player Profiles
```

or

```suggestion
        /// Get Players
```

File: maturity-level-one/README.md [hunk 3]
```diff
@@ -150,20 +150,28 @@ Every API should have documentation in the OpenAPI format. If you want to genera
...
+        /// </summary>
+        /// <param name="topPlayersOnly">Indicates whether to return the top players only</param>
+        /// <remarks>Operation description here</remarks>
```
Comment (tomkerkhove) [inline maturity-level-one/README.md]: This should be a decent, but longer, description of what the operation does.

```suggestion
        /// <remarks>Provides a profile for all known players</remarks>
```

File: maturity-level-one/README.md (hunk 1 above)
Comment (tomkerkhove) [inline maturity-level-one/README.md]: I would re-add an example of the body and where the parameters come in such as:
```csharp
public IHttpActionResult GetPlayers(bool topPlayersOnly)	
{	
    return Ok();	
}
´´´

This indicates that OperationId is composed of `{Controller}_{OperationName}`

File: maturity-level-one/README.md [hunk 4]
```diff
@@ -150,20 +150,28 @@ Every API should have documentation in the OpenAPI format. If you want to genera
...
+| Type | Value | Open Api field | Notes |
+|:--------|:-------------------|:-------------------|:------|
+| XML comment | Summary | Summary | Short description. Human readable, business oriented |
+| XML comment| Remarks | Description | Long description. Human readble, business oriented |
+| XML comment | Returns | N/A | Not used |
+| XML comment | Param name | Parameters/Name | - |
+| XML comment | Param value | Parameters/Description | - |
+| Http Attribute | Name | OperationId | - |
+| SwaggerResponse Attribute | StatusCode | Responseses/{StatusCode} | - |
+| SwaggerResponse Attribute | Description | Responses/Description |  Technical description. Describe whether a body is returned and any particular header.|
```
Comment (tomkerkhove) [inline maturity-level-one/README.md]: Maybe interesting to mention the `Required` attribute which is used to determine if parameters are required or optional.

This used to be required by default, but is optional by default nowadays.

File: maturity-level-one/README.md (hunk 2 above)
Comment (MassimoC) [reply maturity-level-one/README.md]: This is not the operation name, it's the description

File: maturity-level-one/README.md (hunk 1 above)
Comment (MassimoC) [reply maturity-level-one/README.md]: ok, agree

File: maturity-level-one/README.md (hunk 2 above)
Comment (tomkerkhove) [reply maturity-level-one/README.md]: It's actually not, summary is operation name (not to confuse with operation id).
Think the sample is running on an old version of Swashbuckle then.

See https://blog.tomkerkhove.be/2018/04/13/changes-to-azure-api-management-openapi/

**Pull request merged**